cmd/picsv/main.go

```diff
@@ -0,0 +1,323 @@
+					return row[idIndex]
+				}
+			default:
+				return nil, nil, nil, errors.Errorf("unknown IDType: %s", config.IDType)
+			}
+			continue
+		}
+
+		var valGetter func(val string) interface{}
+		srcField, ok := config.SourceFields[fieldName]
+		if !ok {
+			srcField = SourceField{
+				TargetField: fieldName,
+				Type:        "string",
+			}
+			config.SourceFields[fieldName] = srcField
+		}
+		pilosaField, ok := config.PilosaFields[srcField.TargetField]
+		if !ok {
+			pilosaField = Field{
+				Type:      "set",
+				CacheType: pilosa.CacheTypeRanked,
+				CacheSize: 100000,
+				Keys:      true,
+			}
+			config.PilosaFields[fieldName] = pilosaField
+		}
+
+		fieldName = srcField.TargetField
+		switch srcField.Type {
+		case "ignore":
+			continue
+		case "int":
+			valGetter = func(val string) interface{} {
+				intVal, err := strconv.ParseInt(val, 10, 64)
+				if err != nil {
+					return nil
+				}
+				return intVal
+			}
+			fields = append(fields, index.Field(fieldName, pilosaField.MakeOptions()...))
+		case "float":
+			if srcField.Multiplier != 0 {
+				valGetter = func(val string) interface{} {
+					floatVal, err := strconv.ParseFloat(val, 64)
+					if err != nil {
+						return nil
+					}
+					return int64(floatVal * srcField.Multiplier)
+				}
+			} else {
+				valGetter = func(val string) interface{} {
+					floatVal, err := strconv.ParseFloat(val, 64)
+					if err != nil {
+						return nil
+					}
+					return int64(floatVal)
+				}
+			}
+			fields = append(fields, index.Field(fieldName, pilosaField.MakeOptions()...))
+		case "string":
+			valGetter = func(val string) interface{} {
+				if val == "" {
+					return nil // ignore empty strings
+				}
+				return val
+			}
+			fields = append(fields, index.Field(fieldName, pilosaField.MakeOptions()...))
+		case "uint64":
+			valGetter = func(val string) interface{} {
+				uintVal, err := strconv.ParseUint(val, 0, 64)
+				if err != nil {
+					return nil
+				}
+				return uintVal
+			}
+			fields = append(fields, index.Field(fieldName, pilosaField.MakeOptions()...))
+		}
+		header[fieldName] = valueMeta{
+			valGetter:   valGetter,
+			srcIndex:    i,
+			recordIndex: len(fields) - 1,
+		}
+	}
+
+	return fields, header, getIDFn, nil
+}
+
+func main() {
+	if err := commandeer.Run(NewMain()); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func NewConfig() *Config {
+	return &Config{
+		PilosaFields: make(map[string]Field),
+		SourceFields: make(map[string]SourceField),
+		IDType:       "string",
+	}
+}
+
+type Config struct {
+	PilosaFields map[string]Field       `json:"pilosa-fields"`
+	SourceFields map[string]SourceField `json:"source-fields"`
+
+	// IDField denotes which field in the source should be used for Pilosa record IDs.
+	IDField string `json:"id-field"`
+
+	// IDType denotes whether the ID field should be parsed as a string or uint64.
+	IDType string `json:"id-type"`
+}
+
+type Field struct {
+	Type      string           `json:"type"`
```

Comment:
Using the term "type" causes some problems in some java config file parsers/tools because it is a reserved word.  Just a minor annoyance to consider